lib/services/spatial.js

```diff
@@ -36,8 +36,8 @@ class SpatialService {
       var fourfour = dataset.body.id;
       return layer.columns.map((col) => [fourfour, col]);
     }), true);
-
-    return async.map(colSpecs, core.addColumn.bind(core), (err, colResponses) => {
+    //only make 3 requests in parallel. core will die if we do a bunch.
+    return async.mapLimit(colSpecs, 3, core.addColumn.bind(core), (err, colResponses) => {
```

Comment:
it limits the concurrency of the requests. core will lock datasets for mutation operations, and it has a set number of locks it can use, so if you fire 20 requests in parallel it will run out of locks and return 400's